### PR TITLE
[v1.14.x] fabtests/ubertest: cherry-picked fixes for 1.14

### PR DESCRIPTION
**Pull request by aingerson**

### CODE REVIEW

File: fabtests/ubertest/cq.c [hunk 1]
```diff
@@ -71,11 +71,14 @@ bool ft_generates_rx_comp(void)
...
-	    !(is_msg_func(test_info.class_function) &&
-	      test_info.msg_flags & FI_REMOTE_CQ_DATA))
-		return false;
+	if ((test_info.test_class & FI_RMA)) {
+		if (!is_data_func(test_info.class_function) &&
+		    !(is_msg_func(test_info.class_function) &&
+		      test_info.msg_flags & FI_REMOTE_CQ_DATA))
+			return false;
+		if (!(test_info.mode & FI_RX_CQ_DATA))
+			return true;
```
Comment (j-xiong) [inline fabtests/ubertest/cq.c:80]: This returns true for any RMA operation. Should be put inside the body of the previous `if` statement.

Comment (aingerson) [reply fabtests/ubertest/cq.c:80]: We only get here if it is a an RMA operation with CQ data. The check above always returns false for non CQ data operations. If it is an RMA with CQ data and we are not using the RX CQ DATA mode, then we should always write a completion

Comment (j-xiong) [reply fabtests/ubertest/cq.c:80]: my bad, it is fine.

**Comment by j-xiong** [comment]
The main branch has the same code, is the fix needed there? Have you figured out why the main branch didn't hang?

**Comment by j-xiong** [comment]
Ok I saw the other PR. But the second part of the question is still valid :)

**Comment by aingerson** [comment]
@j-xiong :) Yes, main branch didn't hang because of commit dbf437f5c0ea13ff8d24c36519f3342ff2b9ec74 which changed the shm all.test to build a quick test (5 iterations vs 10k). When run with 10k iterations, it hangs on the main branch as well. So what was happening was that ubertest was not expecting cq completions so it was never reading them but shm was writing them so after a while, the CQ got full and we couldn't progress anymore.

**Comment by j-xiong** [comment]
Thanks. Was the change to quick test motivated by working around this issue or for something else (should the long test be restored)? The commit message only talks about fixing the parameter bug.

**Review by j-xiong — APPROVED** [review]

**Comment by aingerson** [comment]
@j-xiong The change I think was motivated by some additions to ubertest that added more combinations so the iterations were reduced in order to add more tests. The change that triggered this issue came later but the issue was not caught because the iterations had already been reduced.

**Pull request merged**